pos_stock_available_online/static/src/js/Screens/ProductScreen/ProductsWidget.esm.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module **/
+
+import ProductsWidget from "point_of_sale.ProductsWidget";
+import Registries from "point_of_sale.Registries";
+
+const StockProductsWidget = (ProductsWidget) =>
+    class StockProductsWidget extends ProductsWidget {
+        setup() {
+            super.setup();
+            this.env.services.bus_service.addChannel(this._getChannelName());
+            this.env.services.bus_service.addEventListener(
+                "notification",
+                this._onNotification.bind(this)
+            );
+        }
+        _getChannelName() {
+            return JSON.stringify([
+                "pos_stock_available_online",
+                String(this.env.pos.config.id),
+            ]);
+        }
+        _onNotification({detail: notifications}) {
+            var payloads = [];
+            for (const {payload, type} of notifications) {
+                if (type === "pos.config/product_update") {
+                    payloads.push(payload);
+                }
+            }
+            this._handleNotification(payloads);
+        }
+        async _handleNotification(payloads) {
+            if (this.env.isDebug()) {
+                console.log("Payloads:", payloads);
+            }
+            const db = this.env.pos.db;
+            const ProductIds = [];
+            for (const payload of payloads) {
+                for (const message of payload) {
+                    var product = db.get_product_by_id(message.product_id);
```

Comment:
you could avoid this if you sent from  backend only notifications used on the pos.config

Reply:
If a new product is added to the POS, this product will not be available in the POS until the POS product is read again (for example, until the page is updated), this check is necessary so that the POS does not process information about the product that is available in the POS but not actually displayed (because the pos db is not yet updated)